stag-library/src/main/java/com/vimeo/stag/KnownTypeAdapters.java

```diff
@@ -336,27 +338,36 @@ public static void write(JsonWriter out, char value) throws IOException {
         }
     }
 
-
     /**
      * Helper class for {@link boolean}.
      */
     public static final class PrimitiveBooleanTypeAdapter {
 
+        private static final List<String> VALID_BOOLEAN_AS_STRING = Arrays.asList("true", "false", "1", "0");
+
         public static boolean read(JsonReader in, boolean defaultValue) throws IOException {
             JsonToken peek = in.peek();
             if (peek == JsonToken.NULL) {
                 in.nextNull();
                 return defaultValue;
-            } else if (peek == JsonToken.STRING) {
+            } else if (peek == JsonToken.STRING || peek == JsonToken.NUMBER) {
                 // support strings for compatibility with GSON 1.7
-                return Boolean.parseBoolean(in.nextString());
+                return Boolean.parseBoolean(booleanAsString(in.nextString()));
```

Comment:
Should we not just keep this simple and have it as 

((name != null) && (name.equalsIgnoreCase("true") || name.equals("1")))


![image](https://cloud.githubusercontent.com/assets/13655724/23178978/5e24a104-f892-11e6-9bcc-ed131feef9a2.png)